File: src/main/java/com/dongtronic/diabot/util/Patterns.java
package com.dongtronic.diabot.util;

import java.util.regex.Pattern;

public class Patterns {
  public static Pattern inlineBgPattern = Pattern.compile("^.*_([0-9]{1,3}\\.?[0-9]?)_.*$");
  public static Pattern separateBgPattern = Pattern.compile("^([0-9]{1,3}\\.?[0-9]?)$");
  public static Pattern unitBgPattern = Pattern.compile("^((.*\\s-?)|(-?))?([0-9.]+)\\s?((mmol)|(mg)).*$");

}

File: src/main/java/com/dongtronic/diabot/util/Patterns.kt
package com.dongtronic.diabot.util

import java.util.regex.Pattern

object Patterns {
    val inlineBgPattern = Pattern.compile("^.*_([0-9]{1,3}\\.?[0-9]?)_.*$")!!
    val separateBgPattern = Pattern.compile("^([0-9]{1,3}\\.?[0-9]?)$")!!
    val unitBgPattern = Pattern.compile("^((.*\\s-?)|(-?))?([0-9.]+)\\s?((mmol)|(mg)).*$")!!

}
